supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/AnalysisOfEndpointConnections.java

```diff
@@ -0,0 +1,75 @@
+package de.tum.cit.endpointanalysis;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import com.thoughtworks.qdox.JavaProjectBuilder;
+import com.thoughtworks.qdox.model.JavaAnnotation;
+import com.thoughtworks.qdox.model.JavaClass;
+import com.thoughtworks.qdox.model.JavaMethod;
+
+public class AnalysisOfEndpointConnections {
+
+    /**
+     * This is the entry point of the analysis of server sided endpoints.
+     *
+     * @param args List of files that should be analyzed regarding endpoints.
+     */
+    public static void main(String[] args) {
+        if (args.length == 0) {
+            System.out.println("No files to analyze.");
+            return;
+        }
+        String[] filePaths = args[0].split("\n");
+        String[] serverFiles = Arrays.stream(filePaths).map(filePath -> "../../" + filePath).filter(filePath -> new File(filePath).exists() && filePath.endsWith(".java"))
+                .toArray(String[]::new);
+        analyzeServerEndpoints(serverFiles);
+    }
+
+    private static void analyzeServerEndpoints(String[] filePaths) {
+        final String requestMappingFullName = "org.springframework.web.bind.annotation.RequestMapping";
+        final List<String> httpMethodFullNames = List.of("org.springframework.web.bind.annotation.GetMapping", "org.springframework.web.bind.annotation.PostMapping",
```

Comment:
```suggestion
        final Set<String> httpMethodFullNames = Set.of("org.springframework.web.bind.annotation.GetMapping", "org.springframework.web.bind.annotation.PostMapping",
```
Makes the contains-check faster and you don't need the ordering here, do you?